test/extensions/io_socket/user_space/io_socket_handle_impl_platform_test.cc

```diff
@@ -0,0 +1,63 @@
+#include "envoy/common/platform.h"
+#include "envoy/event/file_event.h"
+
+#include "extensions/io_socket/user_space/io_socket_handle_impl.h"
+
+#include "test/mocks/event/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace IoSocket {
+namespace UserSpace {
+namespace {
+
+using testing::NiceMock;
+
+class MockFileEventCallback {
+public:
+  MOCK_METHOD(void, called, (uint32_t arg));
+};
+
+// Explicitly mark the test failing on windows and will be fixed.
+class BufferedIoSocketHandlePlatformTest : public testing::Test {
+public:
+  BufferedIoSocketHandlePlatformTest() {
+    first_io_handle_ = std::make_unique<IoSocketHandleImpl>();
+    second_io_handle_ = std::make_unique<IoSocketHandleImpl>();
+    first_io_handle_->setPeerHandle(second_io_handle_.get());
+    second_io_handle_->setPeerHandle(first_io_handle_.get());
```

Comment:
Could we provide a public method to create a pair of userspace io handles that are associated with each other?  Also consider making IoSocketHandleImpl's constructor private.

Reply:
done